compiler/src/dotty/tools/dotc/typer/Synthesizer.scala

```diff
@@ -275,11 +275,10 @@ class Synthesizer(typer: Typer)(using @constructorOnly c: Context):
     val useCompanion = cls.useCompanionAsMirror
 
     if cls.isGenericSum(if useCompanion then cls.linkedClass else ctx.owner) then
-      val elemLabels = cls.children.map(c => ConstantType(Constant(c.name.toString)))
+      val elemLabels = cls.subclasses.map(c => ConstantType(Constant(c.name.toString)))
```

Comment:
I am skeptical that this is the right approach. The elements of a sumMirror are the immediate children of a sealed trait, and I believe we should not change that. What we can maybe change is that sealed sub-traits also figure as children of their super-trait.

Reply:
> What we can maybe change is that sealed sub-traits also figure as children of their super-trait.

In fact that does not work either since that could introduce duplication: https://github.com/lampepfl/dotty/pull/11851#issuecomment-804732682